spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -260,11 +291,24 @@ object DistriOptimizer {
         models.mapPartitions(modelIter => {
           val modelCache = modelIter.next()
           parameters.aggregrateGradientPartition()
-          parameters.gradientPartition.div(ev.fromType(finishedModelNum))
+          if (!optimMethod.isInstanceOf[EAMSGD[T]]) {
```

Comment:
Yes, i understand. I mean we can merge this two if into one, i think there are too many "if (optimMethod.isInstanceOf[EAMSGD[T]])" which makes code looks not clean